### ISSUE TITLE
Key schema inferred from SR is registered in KSQL using ROWKEY instead of the flat schema

### ISSUE THREAD

**Issue by spena** [confluentinc/ksql#8489]
**Describe the bug**
I created a stream with a flat schema. The DESCRIBE command shows the key schema is flat. 
```
ksql> create stream t1(k1 int key, v1 int) with (kafka_topic='t1', format='avro' partitions=1);

ksql> describe t1;

Name                 : T1
 Field | Type                   
--------------------------------
 K1    | INTEGER          (key) 
 V1    | INTEGER                      
--------------------------------

ksql> select k1 from t1 emit changes;
...
```
When I create another stream but inferring the SR schema registered above, the key schema becomes a structured ROWKEY
```
ksql> create stream s1 with (kafka_topic='s1', format='avro', key_schema_id=1, value_schema_id=2);

ksql> describe s1;

Name                 : S1
 Field  | Type                                                     
-------------------------------------------------------------------
 ROWKEY | STRUCT<K1 INTEGER> (key) 
 V1     | INTEGER                                                                                          
-------------------------------------------------------------------

ksql> select k1 from s1 emit changes;
Line: 1, Col: 8: SELECT column 'K1' cannot be resolved
```

**Expected behavior**
If the SR schema is not structured, then the KSQL schema should not be part of ROWKEY.

**Comment by mjsax** [comment]
I agree, that the second statement should result in:
```
Name                 : S1
 Field | Type                   
--------------------------------
 K1    | INTEGER          (key) 
 V1    | INTEGER                      
--------------------------------
```

And I think, it's correct to register `record(k1:int)` as schema in the first statement as we do right now.

My reasoning is the following: Assume we do a `CS(k1, k2, v)` it result in `record(k1,k2)` as key schema (this is the only option we have; we cannot use a primitive avro type). And thus, the "reverse" path should be the same. It seems we would get `ROWKEY struct<k1,k2> (key)` for this case right how what would seem incorrect. The behavior to get two columns is also applied for values, so why should it be different for keys?

If users want a single column key, the original statement would be `CS(struct<k1,k2>, v)` and thus should result in `record(record(k1,k2))` key schema IMHO (to align the the behavior above). -- Of course, we should allow users to get `record(k1,k2)` key schema for `CS(struct<k1,k2,k3>)` but this should not be the default. And if they use the non-default in the first CS statement, it is their responsibility to also change the default in the second CS statement to get single column `struct<k1,k2>` as key -- otherwise, they would get two key columns `(k1,k2)` obviously for key schema `record(k1,k2)` but I think that's ok.

**Comment by vcrfxia** [comment]
I agree it's confusing that the default behavior for key and value inference right now do not match, and that we should update one to match the other. It's not so clear to me that we should update the key behavior to match the value behavior, though. I think it's worth a discussion.

To provide some historical context: the first version of KSQL did not support single (unwrapped) primitive columns for values, i.e., if a user had a single integer value column not contained in an outer object, then there was no way for the user to access that data with KSQL. As such, `(v1)` always meant `struct<v1>`, and schema inference for `struct<v1, v2>` always resulted in `(v1, v2)`. When we later added support for single unwrapped value columns, we had to distinguish between the two with the special keywords `wrap_single_value` and `unwrap_single_value`, which was clunky but worked. Then we introduced support for multiple key columns, and we had a choice to make. We could either introduce `wrap_single_key` and `unwrap_single_key` in order to parallel what existed for values, or we could redesign from scratch and do without those by choosing to unwrap all keys -- an option that wasn't available for values since the first version of KSQL defaulted to wrapped values. We chose the latter since we believed it was simpler, and because users with single key columns are more likely to have primitive key columns rather than a wrapped key column. We chose this option despite knowing the confusion that would result from the default key behavior differing from the default value behavior.

All that to say, that's why we are where we are today. As for how to move forward, the way I see it there are two valid ways schemas that could be inferred from a key schema with two columns: both `(struct<k1, k2>)` and `(k1, k2)` are valid. Today, ksqlDB always chooses the former. I agree it'd be nice for users to have a way to configure which of the two results. We could achieve this by introducing additional properties, which may or may not overlap with the `wrap_single_key/value` properties (they're conceptually similar but not quite the same). Before we do this, we need to complete support for multiple key columns, e.g., for joins (https://www.confluent.io/blog/ksqldb-0-15-reads-more-message-keys-supports-more-data-types/#join-on-sources), otherwise it's not useful for users to have `(k1, k2)`. Note that functionality is already complete for `(struct<k1, k2>)` today.
